### PR TITLE
chart: support controller manager feature gate

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin <wen.chen@daocloud.io>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
we want to use the feature gate `Propagate dependent`, we have not a swith to open the feature.

image

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
please correct me if not

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by calvin0327** [comment]
@Poor12 PTAL

File: charts/karmada/templates/_helpers.tpl [hunk 1]
```diff
@@ -390,4 +390,18 @@ Return the proper karmada kubectl image name
...
+                    {{- $featureGatesFlag = cat $featureGatesFlag $key "=" $value ","  -}}
+               {{- end -}}
+          {{- end -}}
+
+          {{- if gt (len $featureGatesFlag) 0 }}
+               {{- $featureGatesFlag := trimSuffix "," $featureGatesFlag  | nospace -}}
+               {{- printf "%s=%s" "--feature-gates" $featureGatesFlag -}}
+          {{- end -}}
+     {{- end -}}
+{{- end -}}
```
Comment (Poor12) [inline charts/karmada/templates/_helpers.tpl:407]: Add a new line here.

Comment (calvin0327) [reply charts/karmada/templates/_helpers.tpl]: done

File: charts/karmada/values.yaml [hunk 2]
```diff
@@ -285,6 +285,10 @@ controllerManager:
     rollingUpdate:
       maxUnavailable: 0
       maxSurge: 50%
+  ## @param featureGate to controllerManager
+  featureGates:
+    ## @param PropagateDeps is a feature gate for the controllerManager to allow propagate dependent respurce to workloads.
+    PropagateDeps: false
```
Comment (Poor12) [inline charts/karmada/values.yaml:291]: I am thinking that propagateDeps=false is a reasonable default value. How about setting the featureGates to empty by default?

Comment (calvin0327) [reply charts/karmada/values.yaml:291]: Empty is same effect with false by default. I think it's better to set a exact value to controller manager. 
the user can yet know the current value whart is.

Comment (Poor12) [reply charts/karmada/values.yaml:291]: It seems make sense.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**